### PR TITLE
tx_origin_access_control

### PR DESCRIPTION
**Pull request by scab24**
New detector tasked with verifying if there are tx_origin_access_control statements in the code

### CODE REVIEW

File: aderyn_core/src/detect/low/tx_origin_access_control.rs [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+}
+
+#[cfg(test)]
+mod template_detector_tests {
+    use super::TxOriginAccesControl;
+    use crate::detect::detector::{detector_test_helpers::load_contract, IssueDetector};
+
+    #[test]
+    fn test_template_detector() {
+        let context = load_contract(
```
Comment (TilakMaddy) [inline aderyn_core/src/detect/low/tx_origin_access_control.rs:77]: Can you remove some of this `template` stuff and replace it with `tx_origin_detector_tests` (or something like that) ?

Comment (scab24) [reply aderyn_core/src/detect/low/tx_origin_access_control.rs]: I just updated it, but I see it still has the same error. 
Is there anything that could be done to fix it?

Comment (TilakMaddy) [reply aderyn_core/src/detect/low/tx_origin_access_control.rs]: There are a bunch of errors but for this PR can you work on only fixing  This error `detect::low::tx_origin_access_control::tx_origin_detector_tests::test_tx_origin_detector`  ?

**Comment by TilakMaddy** [comment]
Dude, run `foundryup` and then `cli/reportgen.sh`

**Comment by scab24** [comment]
> Dude, try running `foundryup` and then `cli/reportgen.sh`

I just did what you told me and it fixed the issue, but other problems appeared. Next, I did:
- `cargo clippy`
- `cargo fmt`
- `./cli/reportgen.sh`

After this, the initial error disappeared, but now I'm getting an error in check Reports

File: tests/contract-playground/src/TxOriginAccessControl.sol [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    address payable private _owner;
+
+    event TransferMade(address indexed sender, address indexed receiver, uint256 amount);
+
+    constructor() {
+        _owner = payable(msg.sender);
+    }
+
+    modifier onlyOwner() {
+        require(tx.origin == _owner, "Caller is not the owner");
```
Comment (alexroan) [inline tests/contract-playground/src/TxOriginAccessControl.sol:14]: Should we try to capture custom error cases too? For example:

```solidity

if (tx.origin != _owner) {
    revert OwnerIsNotCaller();
}
```

Comment (scab24) [reply tests/contract-playground/src/TxOriginAccessControl.sol]: I'm adding a loop for the identifier of if (IfStatement), at the moment this part doesn't recognize the code you proposed to me:

For now, it correctly identifies the error in the require statements, but it doesn't detect it in the IfStatement. This is my code:

- https://github.com/scab24/aderyn/blob/29ac7fa4e8d18fc30a61785f0f3e9d25e0f8a2b8/aderyn_core/src/detect/low/tx_origin_access_control.rs#L51C1-L71

- https://github.com/scab24/aderyn/blob/%40secoalba%40/tests/contract-playground/src/TxOriginAccessControl.sol

**Comment by alexroan** [comment]
Closing due to out of date with base branch

**Pull request closed without merging**